### PR TITLE
Update kubernetes version and fix Ingress

### PR DESCRIPTION
**Pull request by Mubangizi**
# Description
This PR updates the Kubernetes client version from `10` to `23`.
It also fixes the ingress configurations to use **NetworkingAPI** instead of the **ExtentsionsAPI**

## Trello Ticket ID
Link to Trello ticket is [here](https://trello.com/c/mJDgaC8d/1028-fix-ingress-deployment)

## How Can This Be Tested?

**Note:** the update does not work on staging (CIS- cluster), because it has Kubernetes version `v1.18`, but works on all the other clusters we have at the moment.

- Create a project and an app on the other clusters and it should work.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code

### CODE REVIEW

**Comment by inno-asiimwe** [comment]
Nice, to get this working for all the clusters, we can have the staging cluster upgraded to the kubernetes version being used on the prod cluster. The difference between the 2 versions is non breaking. Since it is staging, it should also be safe to do without worrying so much about the workloads.

**Review by MutegekiHenry — APPROVED** [review]
Great work on this @Mubangizi !

File: migrations/versions/5ca7df80b59f_.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
+"""empty message
+
+Revision ID: 5ca7df80b59f
+Revises: 011c3345849a
+Create Date: 2022-05-04 08:53:42.074660
+
+"""
+from alembic import op
```
Comment (e-ian) [inline migrations/versions/5ca7df80b59f_.py:8]: Just curious about how this migration comes in with the K8s version update. Please enlighten me @Mubangizi. Also is it possible to update all staging clusters as Inno suggested for easy testing?

**Comment by Mubangizi** [comment]
I have removed the migration file

**Pull request merged**